modules/javafx.controls/src/main/java/javafx/scene/control/ControlUtils.java

```diff
@@ -161,24 +161,27 @@ public static <S> void updateSelectedIndices(MultipleSelectionModelBase<S> sm, b
         sm.selectedIndices._beginChange();
 
         while (c.next()) {
-            // it may look like all we are doing here is collecting the removed elements (and
-            // counting the added elements), but the call to 'peek' is also crucial - it is
-            // ensuring that the selectedIndices bitset is correctly updated.
-
             sm.startAtomic();
-            final List<Integer> removed = c.getRemoved().stream()
+
+            final List<Integer> removed = new ArrayList<>(c.getRemovedSize());
```

Comment:
I wonder if we should add a check for 0 size here to bypass all this code if nothing is removed (same for added)

Reply:
We certainly could, or maybe use wasRemoved(), but I doubt there will be much impact. I guess it's preferred to keep changes unrelated to the issue minimal, so I'd leave it as it is if everyone's ok with that.

Reply:
These short circuiting operations tend to be worth it only if it's a critical path. The GC will collect the allocations very efficiently these days. I didn't check what this code segment is used for, but unless it's looped somewhere, I doubt there will by any change.